alu_u32/src/lt/stark.rs

```diff
@@ -76,26 +76,89 @@ where
             builder.assert_bool(local.byte_flag[i]);
         }
 
+        // Check the bit decomposition of the top bytes:
+        let top_comp_1: AB::Expr = local
```

Comment:
Computing the bit decomposition of the top bytes every time might be inefficient, but I was not sure how to check the sign bits of the two inputs otherwise. We might consider using sends into the `Bitwise32Chip` to do some of the bitwise arithmetic instead?